src/web/mjs/connectors/templates/Lezhin.mjs

```diff
@@ -68,165 +92,153 @@ export default class Lezhin extends Connector {
         let cdn = data.match(/cdnUrl\s*:\s*['"]([^'"]+)['"]/);
         let user = data.match(/userId\s*:\s*['"](\d+)['"]/);
         let token = data.match(/token\s*:\s*['"]([^'"]+)['"]/);
+        this.requestOptions.headers.set('Authorization', 'Bearer '+token[1]);
         this.cdnURL = cdn ? cdn[1] : this.cdnURL;
         this.userID = user ? user[1] : undefined;
-        this.accessToken = token ? token[1] : this.accessToken;
         if(this.userID) {
             await fetch(this.url + '/adultkind?path=&sw=all', this.requestOptions);
         }
         // force user locale user setting to be the same as locale from the currently used website ...
         // => prevent a warning webpage that would appear otherwise when loading chapters / pages
         return fetch(this.url + '/locale/' + this.locale, this.requestOptions);
     }
-
     async _getMangaFromURI(uri) {
         let request = new Request(uri, this.requestOptions);
         let data = await this.fetchDOM(request, 'div.comicInfo__detail h2.comicInfo__title');
         let id = uri.pathname.match(/comic\/([^/]+)/)[1];
         let title = data[0].textContent.trim();
         return new Manga(this, id, title);
     }
-
-    /**
-     *
-     */
-    _getMangaFromPage( offset, limit ) {
-        offset = offset || 0;
-        limit = limit || 500;
-        let uri = new URL( this.apiURL + '/comics' );
-        uri.searchParams.set( 'offset', offset );
-        uri.searchParams.set( 'limit', limit );
-        uri.searchParams.set( 'country_code', '' );
-        uri.searchParams.set( 'store', 'web' );
-        uri.searchParams.set( 'adult_kind', 'all' );
-        uri.searchParams.set( 'filter', 'all' );
-        //uri.searchParams.set( '_', Date.now() );
-        return fetch( uri.href, this.requestOptions )
-            .then( response => response.json() )
-            .then( data => {
-                if( data.code ) {
-                    throw new Error( data.description );
-                }
-                let mangaList = data.data.map( manga => {
-                    return {
-                        id: manga.alias, // manga.id
-                        title: manga.title
-                    };
-                } );
-                if( data.hasNext ) {
-                    return this._getMangaFromPage( offset + limit, limit )
-                        .then( mangas => mangaList.concat( mangas ) );
-                } else {
-                    return Promise.resolve( mangaList );
-                }
-            } );
+    async _getMangas() {
+        let mangaList = [];
+        for (let page = 0, run = true; run; page++) {
+            let mangas = await this._getMangasFromPage(page);
+            mangas.length > 0 ? mangaList.push(...mangas) : run = false;
+        }
+        return mangaList;
     }
+    async _getMangasFromPage(page) {
+        //https://www.lezhinus.com/lz-api/v2/contents?menu=general&limit=36&offset=0&order=popular
```

Comment:
The comment may be removed, everything can easily be inferred from the code below